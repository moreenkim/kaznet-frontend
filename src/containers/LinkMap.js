// This component takes an object of links
// and returns them as react-router Link objects
import React, { Component } from 'react';
import { Link } from 'react-router-dom';

export default class LinkMap extends Component {
  render() {
    const { links } = this.props;
    const listLinks = [];
    let counter = 1;

    for (const [key, value] of Object.entries(links)) {
      if (key === 'DELETE TASK') {
        listLinks.push(
          <Link to={value} key={key} className="action-link-alert">
            {key}
          </Link>
        );
      } else {
        listLinks.push(
          <Link to={value} key={key} className="action-link">
            {key}
          </Link>
        );
      }

      // Using counter to generate unique keys for the
      // action-link-divider elements
      listLinks.push(
        <p key={counter} className="action-link-divider">
          |
        </p>
      );

      counter++;
    }

    // Removes last action-link-divider
    listLinks.pop();

    return listLinks;
  }
}
