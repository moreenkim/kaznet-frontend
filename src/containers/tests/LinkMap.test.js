// Test LinkMap
import React from "react";
import { shallow, mount } from "enzyme";
import { Router } from "react-router";
import LinkMap from "../LinkMap";
import toJson from "enzyme-to-json";
import createBrowserHistory from "history/createBrowserHistory";

const history = createBrowserHistory();

describe("containers/LinkMap", () => {
  it("renders without crashing", () => {
    const linkItems = {
      test: "/",
      amount: "/"
    };
    shallow(<LinkMap links={linkItems} />);
  });

  it("renders links correctly", () => {
    const linkItems = {
      test: "/",
      amount: "/"
    };

    const wrapper = mount(
      <Router history={history}>
        <LinkMap links={linkItems} />
      </Router>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount;
  });
});
