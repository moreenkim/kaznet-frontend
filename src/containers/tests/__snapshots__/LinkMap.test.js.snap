// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/LinkMap renders links correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <LinkMap
    links={
      Object {
        "amount": "/",
        "test": "/",
      }
    }
  >
    <Link
      className="action-link"
      key="test"
      replace={false}
      to="/"
    >
      <a
        className="action-link"
        href="/"
        onClick={[Function]}
      >
        test
      </a>
    </Link>
    <p
      className="action-link-divider"
      key="1"
    >
      |
    </p>
    <Link
      className="action-link"
      key="amount"
      replace={false}
      to="/"
    >
      <a
        className="action-link"
        href="/"
        onClick={[Function]}
      >
        amount
      </a>
    </Link>
  </LinkMap>
</Router>
`;
